### PR TITLE
[CRB-189] Handle edge case responses from validators

### PR DESCRIPTION
**Pull request by nathanwhit**
In my local testing I've found that in some cases validators will return misleading "errors", which we should identify and handle properly. For instance, sometimes validator nodes will return an `AuthorizationError` if we try to access invalid state, which is misleading. Additionally, a `get_states_by_prefix` request on a nonexistent address may result in an `ResponseAttributeError`, rather than `None`.

### CODE REVIEW

File: src/handler/context.rs [hunk 1]
```diff
@@ -114,13 +114,22 @@ impl<'tx> HandlerContext<'tx> {
...
+                    Ok(state) if !state.is_empty() => {
+                        let (_addr, value) = &state[0];
+                        Self::find_setting(value, key)
+                    }
+                    Ok(state) if state.is_empty() => {
+                        log::debug!("setting not found for key {:?}", key);
+                        Ok(None)
+                    }
+                    Err(ContextError::ResponseAttributeError(_)) => {
+                        log::debug!("setting not found for key {:?}", key);
```
Comment (CAGS295) [inline src/handler/context.rs:128]: Differentiate the log slightly.

**Review by CAGS295 — DISMISSED** [review]

**Comment by CAGS295** [comment]
Associated ticket?

**Review by seandeneen-gluwa — DISMISSED** [review]

**Review by CAGS295 — APPROVED** [review]

**Comment by atodorov** [comment]
> In my local testing I've found that in some cases validators will return misleading "errors", which we should identify and handle properly. For instance, sometimes validator nodes will return an `AuthorizationError` if we try to access invalid state, which is misleading. Additionally, a `get_states_by_prefix` request on a nonexistent address may result in an `ResponseAttributeError`, rather than `None`.

Is this deterministic or totally random? These looks to me like edge cases which should be covered by unit or integration tests but for that we need to be able to reproduce consistently.

**Review by atodorov — CHANGES_REQUESTED** [review]
From what I can tell from the coverage report most of the newly added changes are not covered by any tests at all: 
https://app.codecov.io/gh/gluwa/Creditcoin-Processor-Rust/compare/11/tree/src/handler

**Comment by CAGS295** [comment]
> > In my local testing I've found that in some cases validators will return misleading "errors", which we should identify and handle properly. For instance, sometimes validator nodes will return an `AuthorizationError` if we try to access invalid state, which is misleading. Additionally, a `get_states_by_prefix` request on a nonexistent address may result in an `ResponseAttributeError`, rather than `None`.
> 
> Is this deterministic or totally random? These looks to me like edge cases which should be covered by unit or integration tests but for that we need to be able to reproduce consistently.

I remember being able to reproduce it consistently when an on-chain setting was missing. I don't remember which one though.

**Review by CAGS295 — APPROVED** [review]

**Comment by atodorov** [comment]
@nathanwhit trying to build from this PR failed with:

```
$ docker build -t gluwa/creditcoin-processor-priv:atodorov-testing-2.0 .
Sending build context to Docker daemon  2.479MB
Step 1/8 : FROM ubuntu:bionic
 ---> 5a214d77f5d7
Step 2/8 : RUN apt-get update  && apt-get install -y  curl  gcc  pkg-config  make  cmake  libssl-dev  automake  autoconf  unzip
 ---> Using cache
 ---> a1666b576baa
Step 3/8 : RUN curl -OLsS https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip  && unzip protoc-3.5.1-linux-x86_64.zip -d protoc3  && rm protoc-3.5.1-linux-x86_64.zip
 ---> Using cache
 ---> 1b13aa514728
Step 4/8 : RUN curl https://sh.rustup.rs -sSf > /usr/bin/rustup-init  && chmod +x /usr/bin/rustup-init  && rustup-init -y
 ---> Using cache
 ---> f55b6333fe27
Step 5/8 : ENV PATH=$PATH:/protoc3/bin:/root/.cargo/bin     CARGO_INCREMENTAL=0
 ---> Using cache
 ---> 2481cb6904e4
Step 6/8 : WORKDIR /ccprocessor-rust
 ---> Using cache
 ---> cc9450a51207
Step 7/8 : COPY . /ccprocessor-rust
 ---> 2578416525bf
Step 8/8 : RUN cd /ccprocessor-rust  && echo "\033[0;32m--- Building ccprocessor-rust ---\n\033[0m"  && rm -rf ./bin/  && mkdir -p ./bin/  && cargo build --release  && cp ./target/release/ccprocessor-rust ./bin/ccprocessor-rust  && cargo build --release --features 'old-sawtooth'  && cp ./target/release/ccprocessor-rust ./bin/ccprocessor-rust-1.7
 ---> Running in 6cc9a767a6d0
--- Building ccprocessor-rust ---

    Updating crates.io index
    Updating git repository `https://github.com/nathanwhit/sawtooth-sdk-rust`
error: failed to get `sawtooth-sdk-creditcoin` as a dependency of package `ccprocessor-rust v0.1.0 (/ccprocessor-rust)`

Caused by:
  failed to load source for dependency `sawtooth-sdk-creditcoin`

Caused by:
  Unable to update https://github.com/nathanwhit/sawtooth-sdk-rust?branch=main#4dfd2a84

Caused by:
  object not found - no match for id (4dfd2a84485aace2f0cd754efa26ae40b3e80efc); class=Odb (9); code=NotFound (-3)
The command '/bin/sh -c cd /ccprocessor-rust  && echo "\033[0;32m--- Building ccprocessor-rust ---\n\033[0m"  && rm -rf ./bin/  && mkdir -p ./bin/  && cargo build --release  && cp ./target/release/ccprocessor-rust ./bin/ccprocessor-rust  && cargo build --release --features 'old-sawtooth'  && cp ./target/release/ccprocessor-rust ./bin/ccprocessor-rust-1.7' returned a non-zero code: 101
```

Do I need to rebase to latest dev before building ?

**Comment by nathanwhit** [comment]
@atodorov I rebased, so it should be working now.

**Review by atodorov — APPROVED** [review]

**Review by CAGS295 — APPROVED** [review]
